server/src/main/java/org/elasticsearch/cluster/metadata/MetadataMappingService.java

```diff
@@ -360,4 +368,87 @@ public TimeValue ackTimeout() {
                     }
                 });
     }
+
+    private Map<String, Object> populateColumnPositions(CompressedXContent mappingUpdateSource,
+                                                        CompressedXContent existingSource,
+                                                        Index index,
+                                                        ClusterState currentState) {
+        Map<String, Object> updatedSourceMap = XContentHelper.convertToMap(mappingUpdateSource.compressedReference(), true).v2();
+        Map<String, Object> existingSourceMap = null;
+        if (existingSource != null) {
+            existingSourceMap = XContentHelper.convertToMap(existingSource.compressedReference(), true).v2();
+        }
+        try {
+            String partitionName = PartitionName.templateName(index.getName());
+            IndexTemplateMetadata indexTemplateMetadata = currentState.metadata().templates().get(partitionName);
+            populateColumnPositions(
+                Maps.getOrDefault(updatedSourceMap, "default", Map.of()),
+                findMaxColumnPosition(Maps.getOrDefault(
+                    // maxColumnPosition of template mappings cannot be used here because an index is a subset of the template
+                    // existingSourceMap contains the latest column positions, if unavailable use updatedSourceMap
+                    existingSourceMap == null ? updatedSourceMap : existingSourceMap,
+                    "default",
+                    Map.of()
+                )),
+                // if partitioned, template-mapping should contain the latest column positions
+                indexTemplateMetadata.mappings().get(Constants.DEFAULT_MAPPING_TYPE)
+            );
+        } catch (Exception e) {
```

Comment:
This could use some explanation. Why is it okay for the above block to fail?

Reply:
Can use `IndexParts.isPartitioned` to see if the index name pattern indicates it's a partition. If so the template must exist.